pkg/proxy/ipvs/graceful_termination_test.go

```diff
@@ -400,3 +401,36 @@ func Test_GracefulDeleteRS(t *testing.T) {
 		})
 	}
 }
+
+func Test_RaceTerminateRSList(t *testing.T) {
```

Comment:
So does running this test without the fix result in a panic?

Reply:
Yes, if not fixed, running this test will cause panic.
```shell
cyclinder@CycLinder-3 ~/Desktop/code/kubernetes/pkg/proxy/ipvs$ go test -race -timeout 60s -run ^Test_RaceTerminateRSList$  -v                                         ✹ ✭fix_concurrent_map 
=== RUN   Test_RaceTerminateRSList
--- PASS: Test_RaceTerminateRSList (0.00s)
==================
WARNING: DATA RACE
Write at 0x00c0001a85a0 by goroutine 21:
  runtime.mapassign_faststr()
      /Users/cyclinder/Downloads/goland/go1.17/src/runtime/map_faststr.go:202 +0x0
  k8s.io/kubernetes/pkg/proxy/ipvs.(*graceTerminateRSList).add()
      /Users/cyclinder/Desktop/code/kubernetes/pkg/proxy/ipvs/graceful_termination.go:66 +0x2a4
  k8s.io/kubernetes/pkg/proxy/ipvs.Test_RaceTerminateRSList.func1()
      /Users/cyclinder/Desktop/code/kubernetes/pkg/proxy/ipvs/graceful_termination_test.go:412 +0x4f

```

Reply:
thanks!